### PR TITLE
drivers: usb: Develop Synopsis USB 2.0 host controller driver

### PR DESCRIPTION
**Pull request by nareshrkumar**
- USB host controller driver for Synopsis DWC2 IP is created
- Required changes to USB host stack have been made to do auto enumeration and register classes
- Mass Storage Class driver for USB 2.0 host is created and integrated with FAT32 file system, same can be accessed by fs shell application

### CODE REVIEW

**Comment by faloj** [comment]
@nareshrkumar Any news on this PR ?

**Comment by nareshrkumar** [comment]
> @nareshrkumar Any news on this PR ?

Hi @faloj : was busy with other project commitments, but I am planning to work on this PR from late this week. All the development is done, need to resolve some twister issues and we would be good from converting this PR to "ready for review".

**Comment by faloj** [comment]
Ok cool ! I was working on adding usb host support for STM32H7 (and other STM32 with same internal USB peripheral) but the IP used in this part is dwc2 so I will wait for this to be merged.
I may post a PR with some tweaks later on so it also works for STM32.

File: drivers/usb/uhc/uhc_common.c [hunk 1]
```diff
@@ -36,6 +36,13 @@ int uhc_submit_event(const struct device *dev,
 		return -EPERM;
 	}
 
+	/* update status of device connection */
+	if (type == UHC_EVT_DEV_CONNECTED_HS) {
```
Comment (GeorgeCGV) [inline drivers/usb/uhc/uhc_common.c:40]: What about `UHC_EVT_DEV_CONNECTED_FS` and `UHC_EVT_DEV_CONNECTED_LS`?

**Comment by faloj** [comment]
@nareshrkumar Do you still plan to finish the work on this PR ? Do you have an estimation on when you will be able to resume it ? Can I be of any help ?

**Pull request closed without merging**